specs/data_structures.md

```diff
@@ -320,31 +286,178 @@ A proof into an SMT is structured as:
+| name                  | type                                          | description                   |
+| --------------------- | --------------------------------------------- | ----------------------------- |
+| `wrappedTransactions` | [WrappedTransaction](#wrappedtransaction)`[]` | List of wrapped transactions. |
 
- | name           | type                            | description           |
- | -------------- | ------------------------------- | --------------------- |
- | `transactions` | [Transaction](#transaction)`[]` | List of transactions. |
+### WrappedTransaction
 
-TODO define a Transaction format
+| name           | type          | description                                                                                       |
+| -------------- | ------------- | ------------------------------------------------------------------------------------------------- |
+| `index`        | `uint64`      | Index of this transaction in the list of wrapped transactions. This is needed for fraud proofs.   |
+| `transaction`  | `Transaction` | Actual transaction.                                                                               |
+| `messageStart` | `uint64`      | _Optional_. Share index (in row-major order) of first share of message this transaction pays for. |
 
-## MessageData
+### Transaction
+
+| name                    | type                      | description                                                                                                                                                                                                                                                           |
+| ----------------------- | ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| TODO                    |                           |                                                                                                                                                                                                                                                                       |
+| `messageSize`           | `uint64`                  | Size of message this transaction pays a fee for, in `byte`s. If this transaction does not pay a fee for a message, must be `0`.                                                                                                                                       |
+| `messageShareRootsRoot` | [HashDigest](#hashdigest) | Merkle root of message share roots of an optional message that this transaction pays a fee to be included in the current block. Messages are split into shares and committed to here. Large messages can span across rows, which requires more roots to the provided. |
```

Comment:
Why do these roots need to be included in the Tx?

Reply:
Those two fields are optional, and only included if a transaction pays for a message to be included. If we want fees to be consensus-critical (a prerequisite for burning some coins for each unit work a block requires), then we need the transaction sender to commit to this information.